backup-restore/uninstall/uninstall-backup-restore.sh

```diff
@@ -0,0 +1,238 @@
+#!/bin/bash 
+# Performs a full uninstall on Backup and Restore from the cli.
+# Uninstall hub before uninstallin any of the spokes.
+# Make sure you are logged into the correct cluster.
+# Execute on a Linux with Bash
+
+USAGE="Usage: $0 [-u] [-s] [-b <Bakup and Restore Name Space>]
+       -u to unistall a spoke installation before uninstalling hub 
+       -s to skip creation of DeleteBackupRequest"
+
+NAMESPACE=ibm-backup-restore
+FORCE=false
+SKIP=false
+
+err_exit()
+{
+        echo "ERROR: $@" >&2
+        exit 1
+}
+
+#SYS_OS=$(uname -s)
+#[ "$SYS_OS" != "Linux" ] && err_exit "$0 should be executed on a Linux system"
+
+
+while getopts "sub:" OPT
+do
+  # echo "ARGS = $OPT : $OPTARG"
+  case ${OPT} in
+    b )
+      NAMESPACE="${OPTARG}"
+      ;;
+    s )
+      SKIP=true
+      ;;
+    u )
+      FORCE=true
+      ;;
+    \? )
+      err_exit "$USAGE"
+      ;;
+    : )
+      err_exit "$USAGE"
+      ;;
+  esac
+done
+
+check_cmd ()
+{
+   (type $1 > /dev/null) || err_exit "$1  command not found"
+}
+
+check_cmd oc
+check_cmd jq
+check_cmd envsubst
+
+oc whoami > /dev/null || err_exit "Not logged in a cluster"
+
+
+print_heading()
+{
+  echo
+  echo -e "===================================================================================================="
+  echo -e "$@"
+  echo -e "===================================================================================================="
+}
+
+
+# Track time to set future expectations
+UNINSTALL_STARTED="$(date +%s)"
+SPOKE=false
+
+ISF_NS=$(oc get spectrumfusion -A --no-headers | cut -d" " -f1)
+[ -z "$ISF_NS" ] &&  ISF_NS=$NAMESPACE
+export ISF_NS
+
+CONNECTION=$(oc -n $NAMESPACE get cm guardian-configmap -o custom-columns="CONN:data.connectionName" --no-headers)
+if [ -n "$CONNECTION" ]
+  then
+     CONN_STAT=$(oc -n ${ISF_NS} get connection $CONNECTION -o custom-columns=S:status.connectionFromRemoteClusterHealth.status --no-headers)
+     SPOKE=true
+     HUB_STATUS=$(oc -n $NAMESPACE get dataprotectionagent -o custom-columns=H:status.hubStatus --no-headers)
+     if [ "$HUB_STATUS" == "Found" ] && [ "$FORCE" != "true" ]
+      then
+         err_exit 'Hub exist, uninstall hub first or use -u option to force uninstall spoke'
+     fi
+fi
+
+if [ "$SKIP" != true ]
```

Comment:
Do you think we should prompt if we are going to delete the backups , for example, you are about to delete all of your Fusion Backup & Restore service backups and will not be able to restore data that was previously backed-up, do you want to proceed ?